### DISCUSSION TITLE
Using custom fonts

### DISCUSSION THREAD

**Discussion by RomainYeye** [chakra-ui/chakra-ui#3457] [Q&A]
Hello guys!

I'm trying to use custom fonts inside my React app, 

I have created a Fonts component looking like this

`import { Global } from "@emotion/react";

export const Fonts = () => (
  <Global
    styles={`
      @font-face {
        font-family: 'Odin Rounded';
        font-style: normal;
        font-weight: 700;
        font-display: swap;
        src: local('Odin Rounded'), url(./fonts/odin-rounded.bold.otf) format('otf');
      }
      `}
  />
)`

Here is what I put inside my App component 

`const theme = extendTheme({
  fonts: {
    heading: 'Odin Rounded',
    body: 'Raleway',
  },`

`<ChakraProvider theme={theme}>
      <Fonts />`

But it's not working. Do you guys have an idea?

**Comment by zlever01** [comment]
@RomainYeye Did you ever figure it out? I'm having the same issue.

**Comment by nikolovlazar** [comment]
I've got custom fonts set up here: [https://github.com/chakra-ui/design-vs-dev/tree/main/challenges/afrosonic-concept-dashboard](https://github.com/chakra-ui/design-vs-dev/tree/main/challenges/afrosonic-concept-dashboard)

Are you looking to register a font face that you'll serve locally?

**Reply by zlever01** [reply]
Thanks for sharing. I tried your setup and still couldn't get custom fonts to work. I think my problem is ssr preventing custom themes.

**Comment by dmitriifrlv** [comment]
@zlever01 I use chakra for my next.js app and I have working custom font styles. What I did:
1) Installed my font style with [npm](https://fontsource.org/)
2) Added them to my custom theme:
```
const theme = extendTheme({
  fonts: {
    heading: "baloo-thambi-2",
    body: "baloo-thambi-2",
  },
```
3) Provided `ChakraProvider` component with this theme and wrapped with it the whole app inside `_app.tsx`.

That's it, custom fonts are added to SSR-components. 

[Excellent guide on how to add custom fonts.](https://chakra-ui.com/guides/using-fonts#option-1-install-with-npm)

**Reply by zlever01** [reply]
Thanks @dmitriifrlv, still not working for me. What sort of works is this setup:

*fonts.js*
```
const Fonts = () => (
    <style jsx global>{`
    @import url('https://fonts.googleapis.com/css2?family=M+PLUS+Rounded+1c:wght@300;700&display=swap');
  `}</style>
)
export default Fonts

```

*theme.js*
```
import { extendTheme } from '@chakra-ui/react'
import { mode } from '@chakra-ui/theme-tools'

const fonts = {
    heading: "'M PLUS Rounded 1c'",
    body: "'M PLUS Rounded 1c'",
}

const theme = extendTheme({ fonts})
export default theme
```

With that setup, I have to turn off chakra-fonts-header and chakra-fonts-body in the inspector before the custom fonts will work. Any idea on why the chakra-fonts are not updating?
<img width="354" alt="Screen Shot 2021-11-08 at 11 08 50 AM" src="https://user-images.githubusercontent.com/51885200/140776991-0bec536a-3367-40e4-a456-536e9450bb9b.png">

**Reply by dmitriifrlv** [reply]
I've tried to recreate the same behaviour but what I passed inside `extendTheme` overwrote the global variables. However, I came up with one more solution for adding custom fonts for next.js:

1) Installing the desired font from [fontsource](https://fontsource.org/fonts)
2) Adding the required font-styles to the file with `ChakraProvider`, eg:
```
import "@fontsource/baloo-thambi-2/500.css";
import "@fontsource/baloo-thambi-2";
```
3)  This step is different from the guide in docs. In `_app.tsx` file we create a component-wrapper for the whole app (`<AppContainer>`) with `styled` from `@emotion/react` and set font-style inside it:

```
//_app.tsx

import styled from "@emotion/styled";

const AppContainer = styled.div`
  font-family: "Baloo Thambi 2";
`;

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <Chakra cookies={pageProps.cookies}>
      <AppContainer>
        <Component {...pageProps} />
      </AppContainer>
    </Chakra>
  );
}
```

Global css-variables for fonts(`--chakra-fonts-...`) are still there but they are overwritten now by this wrapper component.

**Reply by zlever01** [reply]
@dmitriifrlv Thanks! Your wrapper worked.

**Reply by toffyui** [reply]
@dmitriifrlv

Thank you so much! You saved me lots time XD

**Reply by aaarichter** [reply]
thanks @dmitriifrlv  - the odd question is why isn't the custom one overwritten by the chakra provider
